### PR TITLE
migrating `ApiChanges`

### PR DESCRIPTION
**Pull request by vasanth9**
Related #5309 

created new branch and resolved all comments from #5344

### CODE REVIEW

**Comment by AugustinMauroy** [comment]
Our component:
Capture d’écran 2023-05-03 à 20 14 53
nodejs.dev render:
Capture d’écran 2023-05-03 à 20 15 10

I'm not sure what makes the difference.

File: styles/table.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      }
+
+      &:hover {
+        background: var(--brand10);
+      }
+    }
+  }
+}
+
+.dark {
```
Comment (AugustinMauroy) [inline styles/table.scss]:
```suggestion
data-theme='dark'] {
```

File: i18n/locales/en.json [hunk 2]
```diff
@@ -1,4 +1,8 @@
 {
+  "components.api.apiChanges.addedIn": "Added in: {version}",
```
Comment (AugustinMauroy) [inline i18n/locales/en.json]: You should put it at bottom on the file

**Comment by mikeesto** [comment]
Sorry, I'm a bit confused. Would you like us to close https://github.com/nodejs/nodejs.org/pull/5344 and you will instead try to merge this branch?

**Comment by ovflowd** [comment]
**This PR got accidentally closed because the branch got deleted. Please feel free to reopen the PR.**

**Comment by vasanth9** [comment]
resolved merge conflicts

**Comment by AugustinMauroy** [comment]
style is wried I will take a look to know why it's render like that

File: types/api.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export interface ApiChange {
+  version: string | string[];
+  'pr-url': string;
+  description: string;
+}
+
+export interface ApiUpdate {
+  type: 'added' | 'removed' | 'deprecated' | 'introduced_in' | 'napiVersion';
```
Comment (HinataKah0) [inline types/api.ts:8]: nit: Shall we make the naming consistent? Maybe `introduced_in` -> `introducedIn`.
Do you know if it is dependent on what data source?

File: types/api.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export interface ApiChange {
+  version: string | string[];
```
Comment (HinataKah0) [inline types/api.ts:2]: Is it possible to have just `string[]` as the type?

**Review by HinataKah0 — COMMENTED** [review]
Thank you for the amazing contributions!

I left some comments for now, let's discuss. 😄

File: types/api.ts (hunk 3 above)
Comment (vasanth9) [reply types/api.ts:8]: I don't know, I just migrated the types from https://github.com/nodejs/nodejs.dev/blob/main/src/types/api.ts as is.

Comment (vasanth9) [reply types/api.ts:8]: yes `introducedIn`  is consistent

**Comment by Harkunwar** [comment]
Could you run `npm run test:storybook:snapshot` to generate snapshot of your stories.

**Comment by vasanth9** [comment]
> Could you run `npm run test:storybook:snapshot` to generate snapshot of your stories.

I did it , but it updated 4 snaps

File: types/api.ts (hunk 3 above)
Comment (HinataKah0) [reply types/api.ts:8]: It seems that the code is in this [file](https://github.com/nodejs/nodejs.dev/blob/main/util-node/apiDocsTransformUtils.js#L37-L53).
But I haven't actually understood how it works. 😅

Comment (vasanth9) [reply types/api.ts:8]: Ok

Comment (ovflowd) [reply types/api.ts:8]: Note, it should be `introduced_in`, the reason behind is that this comes from the source docs on https://github.com/nodejs/node, so in this case this should **not** be `introducedIn`.

**Comment by ovflowd** [comment]
> > Could you run `npm run test:storybook:snapshot` to generate snapshot of your stories.
> 
> I did it , but it updated 4 snaps

So it does not create a new snapshot? What snapshot it does update? 🤔

File: components/Api/ApiChanges/index.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+export const Default: Story = {
+  args: {
+    update: {
+      type: 'added',
+      version: [version],
+    },
+    changes: [
+      {
+        version: version,
+        'pr-url': 'https://test.com',
```
Comment (AugustinMauroy) [inline components/Api/ApiChanges/index.stories.tsx]: to have perfect mock
```suggestion
        'pr-url': 'https://github.com/nodejs/node/pulls',
```

**Comment by vasanth9** [comment]
<pre><font color="#A347BA">test: </font>[TEST] Summary of all failing tests
<font color="#A347BA">test: </font>[TEST] FAIL components/Common/Banner/index.stories.tsx
<font color="#A347BA">test: </font>  ● Test suite failed to run
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>    page.goto: Timeout 30000ms exceeded.
<font color="#A347BA">test: </font>    =========================== logs ===========================
<font color="#A347BA">test: </font>    navigating to &quot;http://127.0.0.1:6006/iframe.html&quot;, waiting until &quot;load&quot;
<font color="#A347BA">test: </font>    ============================================================
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>      at defaultPrepare (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:27:14)
<font color="#A347BA">test: </font>      at setupPage (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:79:11)
<font color="#A347BA">test: </font>      at CustomEnvironment.setup (node_modules/@storybook/test-runner/playwright/custom-environment.js:9:5)
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST] FAIL components/Api/ApiChanges/index.stories.tsx
<font color="#A347BA">test: </font>  ● Test suite failed to run
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>    page.goto: Timeout 30000ms exceeded.
<font color="#A347BA">test: </font>    =========================== logs ===========================
<font color="#A347BA">test: </font>    navigating to &quot;http://127.0.0.1:6006/iframe.html&quot;, waiting until &quot;load&quot;
<font color="#A347BA">test: </font>    ============================================================
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>      at defaultPrepare (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:27:14)
<font color="#A347BA">test: </font>      at setupPage (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:79:11)
<font color="#A347BA">test: </font>[TEST]       at CustomEnvironment.setup (node_modules/@storybook/test-runner/playwright/custom-environment.js:9:5)
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST] FAIL components/Article/InlineCode/index.stories.tsx
<font color="#A347BA">test: </font>[TEST]   ● Test suite failed to run
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST]     page.goto: Timeout 30000ms exceeded.
<font color="#A347BA">test: </font>    =========================== logs ===========================
<font color="#A347BA">test: </font>    navigating to &quot;http://127.0.0.1:6006/iframe.html&quot;, waiting until &quot;load&quot;
<font color="#A347BA">test: </font>    ============================================================
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>      at defaultPrepare (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:27:14)
<font color="#A347BA">test: </font>[TEST]       at setupPage (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:79:11)
<font color="#A347BA">test: </font>[TEST]       at CustomEnvironment.setup (node_modules/@storybook/test-runner/playwright/custom-environment.js:9:5)
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST] FAIL components/Common/Dropdown/index.stories.tsx
<font color="#A347BA">test: </font>  ● Test suite failed to run
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>    page.goto: Timeout 30000ms exceeded.
<font color="#A347BA">test: </font>    =========================== logs ===========================
<font color="#A347BA">test: </font>    navigating to &quot;http://127.0.0.1:6006/iframe.html&quot;, waiting until &quot;load&quot;
<font color="#A347BA">test: </font>    ============================================================
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>      at defaultPrepare (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:27:14)
<font color="#A347BA">test: </font>      at setupPage (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:79:11)
<font color="#A347BA">test: </font>      at CustomEnvironment.setup (node_modules/@storybook/test-runner/playwright/custom-environment.js:9:5)
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST] FAIL components/Article/Codebox/index.stories.tsx
<font color="#A347BA">test: </font>[TEST]   ● Test suite failed to run
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST]     page.goto: Timeout 30000ms exceeded.
<font color="#A347BA">test: </font>[TEST]     =========================== logs ===========================
<font color="#A347BA">test: </font>    navigating to &quot;http://127.0.0.1:6006/iframe.html&quot;, waiting until &quot;load&quot;
<font color="#A347BA">test: </font>    ============================================================
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>      at defaultPrepare (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:27:14)
<font color="#A347BA">test: </font>      at setupPage (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:79:11)
<font color="#A347BA">test: </font>      at CustomEnvironment.setup (node_modules/@storybook/test-runner/playwright/custom-environment.js:9:5)
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>FAIL components/Common/ShellBox/index.stories.tsx
<font color="#A347BA">test: </font>[TEST]   ● Test suite failed to run
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST]     page.goto: Timeout 30000ms exceeded.
<font color="#A347BA">test: </font>[TEST]     =========================== logs ===========================
<font color="#A347BA">test: </font>[TEST]     navigating to &quot;http://127.0.0.1:6006/iframe.html&quot;, waiting until &quot;load&quot;
<font color="#A347BA">test: </font>    ============================================================
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>      at defaultPrepare (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:27:14)
<font color="#A347BA">test: </font>[TEST]       at setupPage (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:79:11)
<font color="#A347BA">test: </font>[TEST]       at CustomEnvironment.setup (node_modules/@storybook/test-runner/playwright/custom-environment.js:9:5)
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST] FAIL components/Learn/PreviousNextLink/index.stories.tsx
<font color="#A347BA">test: </font>  ● Test suite failed to run
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>    page.goto: Timeout 30000ms exceeded.
<font color="#A347BA">test: </font>    =========================== logs ===========================
<font color="#A347BA">test: </font>    navigating to &quot;http://127.0.0.1:6006/iframe.html&quot;, waiting until &quot;load&quot;
<font color="#A347BA">test: </font>    ============================================================
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST]       at defaultPrepare (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:27:14)
<font color="#A347BA">test: </font>[TEST]       at setupPage (node_modules/@storybook/test-runner/dist/cjs/setup-page.js:79:11)
<font color="#A347BA">test: </font>      at CustomEnvironment.setup (node_modules/@storybook/test-runner/playwright/custom-environment.js:9:5)
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST] 
<font color="#A347BA">test: </font>[TEST] Test Suites: 7 failed, 16 passed, 23 total
<font color="#A347BA">test: </font>[TEST] Tests:       21 passed, 21 total
<font color="#A347BA">test: </font>[TEST] Snapshots:   21 passed, 21 total
<font color="#A347BA">test: </font>[TEST] Time:        175.254 s
<font color="#A347BA">test: </font>[TEST] Ran all test suites.
<font color="#A347BA">test: </font>[TEST] npx wait-on http://localhost:6006 &amp;&amp; npx test-storybook  exited with code 1
<font color="#A347BA">test: </font>--&gt; Sending SIGTERM to other processes..
<font color="#A347BA">test: </font>[SB] npm run storybook -- --ci exited with code SIGTERM
<font color="#A347BA">test: </font>ERROR: command finished with error: command (/home/vasanth/Desktop/openSource/nodeorg/nodejs.org) npm run test exited (1)
<font color="#F66151">command (/home/vasanth/Desktop/openSource/nodeorg/nodejs.org) npm run test exited (1)</font>

<b> Tasks:    </b><font color="#26A269"><b>0 successful</b></font>, 1 total
<b>Cached:    0 cached</b>, 1 total
<b>  Time:    3m40.895s</b> 
<b>Failed:    </b><font color="#C01C28"><b>//#test</b></font>

<span style="background-color:#C01C28"><font color="#300A24"><b> ERROR </b></font></span><font color="#C01C28"> run failed: command  exited (1)</font>
</pre>






The tests were failing when I ran `npx turbo test`

**Comment by vasanth9** [comment]
> > > Could you run `npm run test:storybook:snapshot` to generate snapshot of your stories.
> > 
> > 
> > I did it , but it updated 4 snaps
> 
> So it does not create a new snapshot? What snapshot it does update? thinking



Hey I ran it again, now it created the new snapshot  for ApiChanges

File: types/api.ts (hunk 4 above)
Comment (vasanth9) [reply types/api.ts:2]: we could do, but I am not sure about it. I need someone to decide it. Thank you.

**Comment by ovflowd** [comment]
**FYI:** We deleted the `major/website-redesign` branch to simplify our release and rollout process. Please feel free to recreate this PR or reopen it with the `main` branch as a target.

Thank you, and we apologise for the inconvenience!

**Comment by vasanth9** [comment]
Ok

**Pull request closed without merging**